modules/miekg/miekg.go

```diff
@@ -745,6 +746,7 @@ func (s *Lookup) doLookup(dnsType uint16, dnsClass uint16, name string, nameServ
 // Expose the inner logic so other tools can use it
 func DoLookupWorker(udp *dns.Client, tcp *dns.Client, dnsType uint16, dnsClass uint16, name string, nameServer string, recursive bool) (Result, zdns.Status, error) {
 	res := Result{Answers: []interface{}{}, Authorities: []interface{}{}, Additional: []interface{}{}}
+	res.Resolver = nameServer
```

Comment:
If you're creating the object as a literal on the line above, you should just add this to the literal.